docs/Time_in_Arbitrum.md

```diff
@@ -4,52 +4,55 @@ title: Time in Arbitrum
 
 Messages in an ArbChain's inbox are processed in order.
 Generally, some time will elapse between the time when a message is put into the inbox (and timestamped) and the time when the contract processes the message and carries out the transaction requested by the message.
 
-If your Solidity contract, running on Arbitrum Rollup, accesses `block.number`, this will give you the ArbChain's current time.
-Because ArbChains operate off-chain, an ArbChain's time might differ a little bit from the current Ethereum block time.
-The ArbChain's clock will never get ahead of the Ethereum block time, but the ArbChain's clock might "run a bit slow".
-In particular, it might be as many as 20 blocks behind the true Ethereum time.
+If your Solidity contract, running on Arbitrum Rollup, accesses `block.number` or `block.timestamp`, this will give you the ArbChain's current block or timestamp respectively.
+Because ArbChains operate off-chain, an ArbChain's block number or timestamp might differ a little bit from the current Ethereum block.
+The ArbChain's clock will never get ahead of the Ethereum's, but the ArbChain's clock might "run a bit slow".
+A rollup chain can configure the maximum amount its clock can differ, but generally it will stay within 20 blocks or 10 minutes
+In particular, a rollup chain can configure how far behind it's clock may fall, but generally it will stay within 20 blocks or 10 minutes.
 
-As you would expect, the block time given by `block.number` will never decrease; it can only increase.
+As you would expect, the results given by `block.number` and `block.timestamp` will never decrease; they can only increase.
 
-If you want to know when the current message (the one that requested the transaction you're currently running) was timestamped, you can get it in Solidity by writing:
+If you want to know when the current message (the one that requested the transaction you're currently running) arrived, you can get it by the following Soldiity calls:
 
-    ArbSys(address(100)).currentMessageTime()
+    ArbSys(address(100)).currentMessageBlock()
+    ArbSys(address(100)).currentMessageTimestamp()
 
-At any time, you can get an upper bound on the true Ethereum block time by using this Solidity call:
+At any time, you can get an upper bound on the true Ethereum block number or timestamp by using these Solidity calls:
 
-    ArbSys(address(100)).timeUpperBound()
+    ArbSys(address(100)).blockUpperBound()
+    ArbSys(address(100)).timestampUpperBound()
 
-This will never be more than 20 blocks ahead of `block.number`.
-The real Ethereum block number will always be between the ArbChain's `block.number` and what timeUpperBound() would return (inclusive).
+This will have the same accurancy guarantee from the ArbChain's configuration described about. For example the `blockUpperBound()` can be configured to never be more than 20 blocks ahead of the current Ethereum block number.
```

Comment:
Typo: should be "above" rather than "about".

Reply:
Fixed